contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -127,6 +137,8 @@ contract TransparentUpgradeableProxy is ERC1967Proxy {
      * non-payability of function implemented through dispatchers while still allowing value to pass through.
      */
     function _requireZeroValue() private {
-        require(msg.value == 0);
+        if (msg.value != 0) {
+            revert TransparentUpgradeableProxyNonPayableFunction(msg.value);
```

Comment:
When a non payable function receives value, it revert without a message. So far we replicated this pattern by reverting without a reason. I'm not sure we want a custom error here ... and if we want a custom error, I don't think it should include the msg.value.

Reply:
I agree with the second point of removing `msg.value`, would be in the transaction itself anyways. However, I disagree with the first point of not having the custom error. I think the very reason we use an error here is to provide the missing context for adding `payable` to the functions.